Scripts/Chunk.gd

```diff
@@ -28,28 +29,30 @@ func _generate_surface() -> Array:
 	var indices := PackedInt32Array()
 	var uvs := PackedVector2Array()
 	var triangle_index: int = 0
+	var mesh_simplification_increment: int = 1 if level_of_detail == 0 else level_of_detail * 2
+	var vertices_per_line: int = (chunk_data.size - 1) / mesh_simplification_increment + 1
```

Comment:
At LODs > 3, the mesh disappears.
This should be fixed.

At worst, if it's impossible to simplify the mesh further, you may still want a minimum of 4 vertices to form a quad.